xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -334,18 +334,25 @@ void CGLContextEGL::Detach()
 bool CGLContextEGL::IsSuitableVisual(XVisualInfo *vInfo)
 {
   EGLConfig config = GetEGLConfig(m_eglDisplay, vInfo);
+  return config != EGL_NO_CONFIG;
+}
+
+bool CGLContextEGL::SuitableCheck(EGLDisplay& eglDisplay, EGLConfig& config)
+{
   if (config == EGL_NO_CONFIG)
```

Comment:
You can move this block to `IsSuitableVisual`

Reply:
Your mean to let the
```cpp
  if (config == EGL_NO_CONFIG)
  {
    CLog::Log(LOGERROR, "Failed to determine egl config for visual info");
    return false;
  }
```
away?

Reply:
And:
```cpp
bool CGLContextEGL::IsSuitableVisual(XVisualInfo *vInfo)
{
  EGLConfig config = GetEGLConfig(m_eglDisplay, vInfo);
  if (config == EGL_NO_CONFIG)
  {
    CLog::Log(LOGERROR, "Failed to determine egl config for visual info");
    return false;
  }
  return true;
}
```

Reply:
If yes makes not direct sense, before was it only to report a failed GetEGLConfig and to prevent the further check. But now becomes everything done inside the GetEGLConfig and can report EGL_NO_CONFIG frequently.

Reply:
Why should `GetEGLConfig` frequently report `EGL_NO_CONFIG`? In that case we will not get any display

Reply:
I have removed the IsSuitableVisual call complete, where this becomes called is the GetEGLConfig called directly after. 

Now think that it is a remnant of the conversion of the old CGLContextGLX where this call was necessary.

The ```CLog::Log(LOGERROR, "Failed to determine egl config for visual info");``` should never come, only for the worst case where a "eglGetConfigs" call returns true with a empty entry in list.